tmc-langs-java/src/main/java/fi/helsinki/cs/tmc/langs/java/maven/MavenInvokatorMavenTaskRunner.java

```diff
@@ -75,4 +93,54 @@ public void consumeLine(String line) {
             throw new MavenExecutorException(e);
         }
     }
+
+    private Path useBundledMaven() {
+        Path mavenHome = getConfigDirectory();
+        if (Files.exists(mavenHome)) {
+            log.info("Maven already extracted");
+
+            // Add the name of the extracted folder to the path
+            return mavenHome.resolve("apache-maven-3.3.9");
+        }
+        log.info("Maven bundle not previously extracted, extracting...");
+        try {
+
+            InputStream data = getClass().getResourceAsStream("apache-maven-3.3.9.zip");
+            Preconditions.checkNotNull(data,
+                "Couldn't load bundled maven from tmc-langs-java.jar.");
+            Path tmpFile = File.createTempFile("tmc-maven", "zip").toPath();
+            Files.copy(data, tmpFile, StandardCopyOption.REPLACE_EXISTING);
+            Archiver archiver = ArchiverFactory.createArchiver(ArchiveFormat.ZIP);
+            archiver.extract(tmpFile.toFile(), mavenHome.toFile());
+
+            // Add the name of the extracted folder to the path
+            return mavenHome.resolve("apache-maven-3.3.9");
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    static Path getConfigDirectory() {
+        Path configPath;
+
+        if (System.getProperty("os.name").toLowerCase().contains("windows")) {
```

Comment:
Are we already dependent on Apache Commons? If yes, use this instead:
http://commons.apache.org/proper/commons-lang/javadocs/api-release/org/apache/commons/lang3/SystemUtils.html#IS_OS_WINDOWS